src/lib/runner.js

```diff
@@ -1,127 +1,206 @@
-import fs from 'fs';
-import os from 'os';
-import path from 'path';
-import { spawn, spawnSync } from 'child_process';
+import fs from 'node:fs/promises';
+import os from 'node:os';
+import path from 'node:path';
+import { spawn } from 'node:child_process';
 
 class Runner {
-  constructor(enableStdOut = false, forwardParentArgs = false) {
-    this.enableStdOut = enableStdOut; // debugging tests
-    this.forwardParentArgs = forwardParentArgs;
-    this.setupFiles = [];
-    this.childProcess = null;
+  #rootDir;
+  #setupFiles = [];
+  #childProcess = null;
+
+  #enableDebug = false;
+  #forwardParentArgs = false;
```

Comment:
I made the instance variables private to enforce encapsulation and so we don't have to deal with unexpected values